core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/app/RemoteResourceImpl.java

```diff
@@ -150,7 +150,8 @@ public boolean verify(String hostname, SSLSession session) {
                 return uc.getInputStream();
 
             } catch (IOException e) {
-                throw new RegistryException("Couldn't open stream to source URL " + contentURL, e);
+                throw new RegistryException("Couldn't open stream to source URL " + contentURL,
+                        new Throwable("Couldn't open stream to source URL " + contentURL));

```

Comment:
we need to log the exception as a debug log at least if we are suppressing it. Also, can't we just throw the RegistryException without having a Throwable e?

Reply:
Please check 139a98b07174a8884ed4920ef9d7341d56736e94